rhizome/host/lib/vm_setup.rb

```diff
@@ -770,4 +755,144 @@ def start_systemd_unit
   def restart_systemd_unit
     r "systemctl restart #{q_vm}"
   end
+
+  def build_ch_service(header:, footer:, slice_name:, mem_gib:, max_vcpus:, cpu_topology:, storage_volumes:, storage_params:, nics:, pci_devices:)
+    disk_params = storage_volumes.map { |volume|
+      if volume.read_only
+        "path=#{volume.image_path},readonly=on"
+      else
+        "vhost_user=true,socket=#{volume.vhost_sock},num_queues=#{volume.num_queues},queue_size=#{volume.queue_size}"
+      end
+    }
+    disk_params << "path=#{vp.cloudinit_img}"
+
+    disk_args =
+      if Gem::Version.new(@ch_version.version) >= Gem::Version.new("36")
+        "--disk #{disk_params.join(" ")}"
+      else
+        disk_params.map { |x| "--disk #{x}" }.join(" ")
+      end
+
+    net_params = nics.map { "--net mac=#{_1.mac},tap=#{_1.tap},ip=,mask=,num_queues=#{max_vcpus * 2 + 1}" }
+    pci_device_params =
+      if pci_devices.empty?
+        ""
+      elsif Gem::Version.new(@ch_version.version) >= Gem::Version.new("36")
+        "\\\n--device #{pci_devices.map { |dev| "path=/sys/bus/pci/devices/0000:#{dev[0]}/" }.join(" ")}"
+      else
+        "\\\n" + pci_devices.map { |dev| "--device path=/sys/bus/pci/devices/0000:#{dev[0]}/" }.join(" ")
```

Comment:
Can you explain the need for `\\\n`? I could understand it if `pci_device_params` were on the same line as `net_params.join(" ")` below, but it isn't.  Seems simpler to have `\\` after the `net_params.join(" ")` below, and not have `\\\n`, but maybe there is a reason it was done this way?

Reply:
yeah, it was on the same line but accidentally moved to a separate line. `pci_device_params` are empty except for the GPU case, i.e., empty most of the time. This code is pretty brittle and I made a small change here. Thanks!